### PR TITLE
Bind environment variables to CLI flags

### LINKED ISSUE #1882: Bind flags to environment variables
## Version of Singularity:

```
$ singularity --version
singularity version 3.0.0-alpha.1-117-gc297513f
```

### Expected behavior

Most if not all of the flags in Singularity can also be controlled via environment variables.  For instance, the `SINGULARITY_BINDPATH` variable allows me to do this.
```
$ ls /data
foo

$ export SINGULARITY_BINDPATH=/data:/mnt

$ singularity exec my_container ls /mnt
foo
```

### Actual behavior

```
$ export SINGULARITY_BINDPATH=/data:/mnt

$ singularity exec my_container ls /mnt
```

### Steps to reproduce behavior

Try using environment variables to control flags.

### PR DESCRIPTION
**Pull request by GodloveD**
**Description of the Pull Request (PR):**

Second stab at a framework for binding environment variables to flags.  This time with some help from @bauerm97.  

**This fixes or addresses the following GitHub issues:**

- fixes #1882 

**Checkoff for all PRs:**

- [x] I have read the [Guidelines for Contributing](https://github.com/singularityware/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- [ ] I have added changes to the [CHANGELOG](https://github.com/singularityware/singularity/blob/development/CHANGELOG.md) and and documentation updates to the [singularityware](https://www.github.com/singularityware/singularityware.github.io) documentation base.
- [ ] I have tested this PR locally with a `make test`
- [x] This PR is against the project's master branch
- [x] I have added myself as a contributor to the [contributors's file](https://github.com/singularityware/singularity/blob/master/CONTRIBUTORS.md)
- [x] This PR is ready for review and/or merge

Attn: @singularityware-admin

### CODE REVIEW

File: src/cmd/singularity/cli/cli_test.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import (
+	"testing"
+
+	"github.com/magiconair/properties/assert"
+	"github.com/spf13/cobra"
+)
+
+var cmd cobra.Command
+
+func TestEnvAppend(t *testing.T) {
```
Comment (ArangoGutierrez) [inline src/cmd/singularity/cli/cli_test.go:18]: if no privs needed to run this test, we should do
```go
	test.DropPrivilege(t)
	defer test.ResetPrivilege(t)
```

File: src/cmd/singularity/cli/cli_test.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	envAppend(cmd.Flag("appendFlag"), "appendval")
+	v = cmd.Flag("appendFlag").Value.String()
+	assert.Equal(t, v, "[appendval]", "The flag should be set to the value provided.")
+
+	envAppend(cmd.Flag("appendFlag"), "appendval")
+	v = cmd.Flag("appendFlag").Value.String()
+	assert.Equal(t, v, "[appendval,appendval]", "The flag should appended with the value provided.")
+}
+
+func TestEnvBool(t *testing.T) {
```
Comment (ArangoGutierrez) [inline src/cmd/singularity/cli/cli_test.go:36]: if no privs needed to run this test, we should do
```go
	test.DropPrivilege(t)
	defer test.ResetPrivilege(t)
```

File: src/cmd/singularity/cli/cli_test.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	cmd.Flags().BoolVar(&boolFlag, "boolFlag", false, "")
+	v := cmd.Flag("boolFlag").Value.String()
+	assert.Equal(t, v, "false", "The flag should be unset.")
+
+	envBool(cmd.Flag("boolFlag"), "any string")
+	v = cmd.Flag("boolFlag").Value.String()
+	assert.Equal(t, v, "true", "The flag should be set to true.")
+}
+
+func TestEnvStringNSlice(t *testing.T) {
```
Comment (ArangoGutierrez) [inline src/cmd/singularity/cli/cli_test.go:50]: if no privs needed to run this test, we should do
```go
	test.DropPrivilege(t)
	defer test.ResetPrivilege(t)
```

**Review by ArangoGutierrez — DISMISSED** [review]

File: src/cmd/singularity/cli/cli_test.go (hunk 1 above)
Comment (bauerm97) [reply src/cmd/singularity/cli/cli_test.go:18]: I think contrarily, if the test doesn't specifically need privilege, I think it's OK to skip this

**Review by bauerm97 — APPROVED** [review]
I think this is in a good state to merge, as long as it's passing Travis. @GodloveD consider this approved to merge

**Comment by ArangoGutierrez** [comment]
go fmt indpect.go (evidently vscode is not doing it's job) (98d17430) 
My young Padawan , remember, is never the tool, is the user...

**Pull request merged**